File: python/nt_solutions/coding_1_approximation/exo9.py
from numpy import linalg

f1 = np.copy(fL)

for i in range(1,n//w+1):
    for j in range(1,n//w+1):
        f1[(i-1)*w: i*w, (j-1)*w: j*w] = idct2(f1[(i-1)*w: i*w, (j-1)*w: j*w])

print("Error |f-f1|/|f| =", linalg.norm(f-f1)/linalg.norm(f))

File: r/nt_solutions/coding_1_approximation/exo9.R





f1 <- fL

for (i in 1:(n%/%w)){
  for (j in 1:(n%/%w)){
    f1[((i-1)*w+1):(i*w), ((j-1)*w+1):(j*w)] <- idct2(fL[((i-1)*w+1):(i*w), ((j-1)*w+1):(j*w)])
  }
}

print(paste("Error |f-f1|/|f| =", norm(as.matrix(f)-f1)/norm(as.matrix(f))))